pkg/tnf/handlers/hugepages/hugepages.go

```
@@ -0,0 +1,173 @@
+	return &Hugepages{
+		timeout: timeout,
+		result:  tnf.ERROR,
+		args: []string{
+			"oc", "get", "machineconfigs", "-l", "machineconfiguration.openshift.io/role=worker",
+			"-o", "custom-columns=KARGS:.spec.kernelArguments",
+			"|", "grep", "-v", "nil", "|", "grep", "-E", "'hugepage|KARGS'",
+		},
+	}
+}
+
+// Args returns the command line args for the test.
+func (hp *Hugepages) Args() []string {
+	return hp.args
+}
+
+// GetIdentifier returns the tnf.Test specific identifiesa.
+func (hp *Hugepages) GetIdentifier() identifier.Identifier {
+	return identifier.HugepagesIdentifier
+}
+
+// Timeout returns the timeout in seconds for the test.
+func (hp *Hugepages) Timeout() time.Duration {
+	return hp.timeout
+}
+
+// Result returns the test result.
+func (hp *Hugepages) Result() int {
+	return hp.result
+}
+
+// ReelFirst returns a step which expects the ping statistics within the test timeout.
+func (hp *Hugepages) ReelFirst() *reel.Step {
+	return &reel.Step{
+		Expect:  []string{hpRegex},
+		Timeout: hp.timeout,
+	}
+}
+
+// ReelMatch sets the hugepages parameters based on cluster configuration and RHEL defaults
+func (hp *Hugepages) ReelMatch(_, _, match string) *reel.Step {
+	trimmedMatch := strings.Trim(match, "\n")
+	lines := strings.Split(trimmedMatch, "\n")[1:] // First line is the headers/titles line
+
+	params := map[string]string{}
+
+	// Each line is of the form [name=value name=value ...]
+	// Find the relevant parameters and store in params
+	for _, line := range lines {
+		line = line[1 : len(line)-1] // trim '[' and ']'
+		fields := strings.Fields(line)
+		for _, field := range fields {
+			nameValue := strings.Split(field, "=")
+			name := nameValue[0]
+			value := nameValue[1]
+			if isHugepagesParam(name) {
+				params[name] = value
+			}
+		}
+	}
+
+	// Use params for determining the hugepages settings
+	hugepages, ok := params["hugepages"]
+	if ok {
+		hp.hugepages, _ = strconv.Atoi(hugepages)
```

Same here.  I am fine with this but just wanted to note in case Charlie takes a look.  The err can be ignored as the output is well known to be an integer.